tests/src/core/importType.js

```diff
@@ -42,20 +43,35 @@ describe('importType(name)', function () {
   })
 
   it("should return 'internal' for non-builtins resolved outside of node_modules", function () {
-    const pathContext = testContext({ "import/resolver": { node: { paths: [ path.join(__dirname, '..', '..', 'files') ] } } })
+    const pathContext = testContext({ "import/resolver": { node: { paths: [pathToTestFiles] } } })
     expect(importType('importType', pathContext)).to.equal('internal')
   })
 
   it.skip("should return 'internal' for scoped packages resolved outside of node_modules", function () {
-    const pathContext = testContext({ "import/resolver": { node: { paths: [ path.join(__dirname, '..', '..', 'files') ] } } })
+    const pathContext = testContext({ "import/resolver": { node: { paths: [pathToTestFiles] } } })
     expect(importType('@importType/index', pathContext)).to.equal('internal')
   })
     
   it("should return 'internal' for internal modules that are referenced by aliases", function () {
-    const pathContext = testContext({ 'import/resolver': { node: { paths: [path.join(__dirname, '..', '..', 'files')] } } })
+    const pathContext = testContext({ 'import/resolver': { node: { paths: [pathToTestFiles] } } })
     expect(importType('@my-alias/fn', pathContext)).to.equal('internal')
   })
 
+  it("should return 'internal' for aliased internal modules that look like core modules (node resolver)", function () {
+    const pathContext = testContext({ 'import/resolver': { node: { paths: [pathToTestFiles] } } })
+    expect(importType('constants/index', pathContext)).to.equal('internal')
+    // the following assertion fails because the node resolver runs an resolve.isCore('constants') without first checking paths config
+    // expect(importType('constants', pathContext)).to.equal('internal')
+  })
+
+  it("should return 'internal' for aliased internal modules that look like core modules (webpack resolver)", function () {
+    const webpackConfig = { resolve: { modules: [pathToTestFiles, 'node_modules'] } }
+    const pathContext = testContext({ 'import/resolver': { webpack: { config: webpackConfig } } })
+    expect(importType('constants/index', pathContext)).to.equal('internal')
+    // the following assertion fails because the webpack resolver runs an resolve.isCore('constants') without first checking paths config
+    // expect(importType('constants', pathContext)).to.equal('internal')
```

Comment:
With my changes, this will still return `'external'` unless changes are made to the webpack resolver.

Reply:
In particular: https://github.com/benmosher/eslint-plugin-import/blob/master/resolvers/webpack/index.js#L41-L43

Reply:
ditto

Reply:
let's uncomment this, at least, just like above.

Reply:
It's a little crude, but moving the coreLibs check until after module resolution in the webpack resolver does appear to be a solution. Added that fix to the PR.

That said, I know webpack has had kind of a weird relationship with core node.js modules, so it's hard to tell if the fix is always going to be correct.

https://webpack.js.org/configuration/node/
https://blog.logrocket.com/new-features-in-webpack-5-2559755adf5e (removing default core polyfills in v5)

Reply:
If that's the case, we'd want to pivot the behavior based on webpack version, i think?

Reply:
Since webpack 5 isn't out yet, I think just moving the coreLibs check works for now.